single-p4_action-resistance_hub.php

```diff
@@ -0,0 +1,360 @@
+<?php
+
+/**
+ * Template Name: Resistance Hub Campaign
+ * The template for displaying Resistance Hub campaign actions.
+ *
+ * @package  WordPress
+ * @subpackage  Timber
+ * @since    Timber 0.1
+ */
+
+$features = get_option('planet4_features');
+
+if (isset($features['action_options']) && (bool) $features['action_options']) {
+    add_filter(
+        'render_block',
+        function ($block_content, $block) {
+            global $post;
+
+            if (
+                isset($block['blockName']) &&
+                $block['blockName'] === 'planet4-blocks/action-button-text' &&
+                $post && $post->post_type === 'p4_action'
+            ) {
+                $action_task_type_value = get_post_meta($post->ID, 'action_task_type', true);
+                $action_deadline_value = get_post_meta($post->ID, 'action_deadline', true);
+
+                if ($action_task_type_value && $action_deadline_value) {
+                    $block_content = preg_replace(
+                        '/(<a[^>]*class="[^"]*btn[^"]*"[^>]*)/',
+                        // phpcs:disable Generic.Files.LineLength.MaxExceeded
+                        '$1 data-tasktype="' . esc_attr($action_task_type_value) . '"' . ' data-deadline="' . esc_attr($action_deadline_value) . '"',
+                        $block_content,
+                        1
+                    );
+                }
+            }
+            return $block_content;
+        },
+        10,
+        2
+    );
+
+    include get_template_directory() . '/single-p4_action.php';
+
+    ?>
+  <style>
```

Comment:
We can move all of this to a file, but it's easier for now to put it all together